src/quorum/datadriven_test.rs

```diff
@@ -0,0 +1,142 @@
+use crate::quorum::{AckIndexer, Index};
+use crate::{default_logger, HashSet, JointConfig, MajorityConfig};
+use datadriven::{run_test, TestData};
+
+#[test]
+fn test_data_driven_quorum() -> anyhow::Result<()> {
```

Comment:
Is the returning value necessary?

Reply:
Yes, we return lots of `Result` in `run_test`, such as `read_to_string`,  `OpenOptions::new()`